runtime/onert/core/src/backend/controlflow/kernel/PermuteLayer.h

```diff
@@ -28,22 +28,40 @@ namespace controlflow
 namespace kernel
 {
 
-class PermuteLayer : public ::onert::exec::IPermuteFunction
+class PermuteLayer : public onert::exec::IPermuteFunction
 {
 public:
-  PermuteLayer() = default;
-
-public:
-  void configure(std::shared_ptr<backend::ITensor> input, std::shared_ptr<backend::ITensor> output,
-                 size_t rank);
+  PermuteLayer(const std::vector<std::shared_ptr<onert::backend::ITensor>> &src_tensors,
+               const std::vector<std::shared_ptr<onert::backend::ITensor>> &dst_tensors,
+               std::vector<size_t> ranks)
+  {
+    assert(src_tensors.size() == dst_tensors.size());
+    assert(src_tensors.size() == ranks.size());
+    _src_tensors = src_tensors;
+    _dst_tensors = dst_tensors;
+    _ranks = ranks;
+  }
 
   void optimize() override
   {
-    if (_src_tensors.at(0) == _dst_tensors.at(0))
+    // Remove copying of tensor as nullptr
+    auto src_it = _src_tensors.begin();
+    auto dst_it = _dst_tensors.begin();
+    auto rank_it = _ranks.begin();
+    while (src_it != _src_tensors.end())
     {
-      _src_tensors.clear();
-      _dst_tensors.clear();
-      _ranks.clear();
+      if ((*src_it == *dst_it) || (*src_it == nullptr || *dst_it == nullptr))
```

Comment:
Can a case where `*src_it == nullptr` but `*dst_it != nullptr` happen? I wonder if it's ok to erase iterators in that case.

Reply:
This condition means that "src tensor and dst tensor are the same" or "src_tensor == nullptr" or "dst_tensor". If a case where `*src_it == nullptr` but `*dst_it != nullptr` happens, this condition is true and then erase those tensors. I think it's okay.